### PR TITLE
add ROCBLAS_PATH to Makefiles

### PR DESCRIPTION
**Pull request by amcamd**
- README files have line 
``` If rocBLAS is not installed you can set the environment variable ROCBLAS_PATH to point to the location of your rocblas build. ```
- The additions to Makefile provide for this functionality.

### CODE REVIEW

**Review by TorreZuk — COMMENTED** [review]
You need to support it correctly when env var not set, so set it to use new include path for ROCBLAS_PATH to ROCM_PATH/include.  We moved away from /opt/rocm/rocblas so should pick up the default which as I mentioned a while back is now confounding as you can't pick subset of packages as all are in /opt/rocm/include.  Thus the order fixes I mention here are required now (they weren't required before in earlier code).

**Review by TorreZuk — APPROVED** [review]
Very good, hopefully builds without warning of duplicate flags for default of no override.  Then we can cherry pick to 5.2 release branch

**Comment by amcamd** [comment]
The following three cases work as expected:
- no rocblas installed, custom rocblas and set ROCBLAS_PATH: uses custom rocblas in ROCBLAS_PATH
- rocblas installed, custom rocblas and set ROCBLAS_PATH: uses custom rocblas in ROCBLAS_PATH
- rocblas installed, ROCBLAS_PATH not set: uses installed rocblas

File: Languages/C/Makefile [hunk 1]
```diff
@@ -43,13 +43,13 @@ CC=gcc
...
 #CC=$(HIPCXX)
 OPT = -ggdb -O0 -march=native -Wall
-INC = -isystem$(ROCM_PATH)/include -I../../common
+INC = -isystem$(ROCM_PATH)/include -isystem$(ROCBLAS_PATH)/include -I../../common
 CCFLAGS = -std=c11 $(INC) $(OPT) 
 ifneq ($(CC),$(HIPCXX))
 	CCFLAGS += -D__HIP_PLATFORM_AMD__ -D__HIP_PLATFORM_HCC__
 endif
 
-LDFLAGS=-L$(ROCM_PATH)/lib -lrocblas -Wl,-rpath=$(ROCM_PATH)/lib -lm -lpthread -lc
```
[the remark this answers is not in the document]
Comment (amcamd) [reply Languages/C/Makefile:52]: Thanks, fixed

File: Level-1/nrm2/Makefile [hunk 2]
```diff
@@ -44,13 +44,13 @@ CXX=g++
 # uncomment to use hip compiler
 #CXX=$(HIPCXX)
 OPT = -g -Ofast -march=native -Wall
-INC = -I$(COMMON_PATH) -isystem$(ROCM_PATH)/include 
+INC = -I$(COMMON_PATH) -isystem$(ROCM_PATH)/include -isystem$(ROCBLAS_PATH)/include
```
[the remark this answers is not in the document]
Comment (amcamd) [reply Level-1/nrm2/Makefile]: Thanks, fixed

File: Level-1/scal/Makefile [hunk 3]
```diff
@@ -44,13 +44,13 @@ CXX=g++
...
 OPT = -g -Ofast -march=native -Wall
-INC = -I$(COMMON_PATH) -isystem$(ROCM_PATH)/include 
+INC = -I$(COMMON_PATH) -isystem$(ROCM_PATH)/include -isystem$(ROCBLAS_PATH)/include
 CXXFLAGS = -std=c++14 $(INC) $(OPT) 
 ifneq ($(CXX),$(HIPCXX))
 	CXXFLAGS += -D__HIP_PLATFORM_AMD__ -D__HIP_PLATFORM_HCC__
 endif
 
-LDFLAGS=-L$(ROCM_PATH)/lib -lrocblas -Wl,-rpath=$(ROCM_PATH)/lib -lm -lpthread -lstdc++
+LDFLAGS=-L$(ROCM_PATH)/lib -L$(ROCBLAS_PATH)/lib -lrocblas -Wl,-rpath=$(ROCM_PATH)/lib -Wl,-rpath=$(ROCBLAS_PATH)/lib -lm -lpthread -lstdc++
```
[the remark this answers is not in the document]
Comment (amcamd) [reply Level-1/scal/Makefile]: Thanks, fixed

**Pull request merged**